lib/ipfs/IpfsStorage.ts

```diff
@@ -83,65 +91,44 @@ export default class IpfsStorage {
    * This method also allows easy mocking in tests.
    */
   private async fetchContent (hash: string, maxSizeInBytes: number): Promise<FetchResult> {
-    // files.getReadableStream() fetches the content from network if not available in local repo and stores in cache which will be garbage collectable.
-    const readableStream = await (this.node as any).getReadableStream(hash);
 
-    let fetchResult: FetchResult = { code: FetchResultCode.Success };
+    const fetchResult: FetchResult = { code: FetchResultCode.Success };
     let bufferChunks: Buffer[] = [];
     let currentContentSize = 0;
-    let resolveFunction: any;
-    let rejectFunction: any;
-
-    const fetchContent = new Promise((resolve, reject) => {
-      resolveFunction = resolve;
-      rejectFunction = reject;
-    });
-
-    readableStream.on('data', (file: any) => {
-      // If content is of directory type, set return code as "not a file", no need to setup content stream listeners.
-      if (file.type === 'dir') {
-        console.info(`Content is of directory type for hash ${hash}, skipping this bad request.`);
-        fetchResult.code = FetchResultCode.NotAFile;
-
-        readableStream.destroy();
-        return;
-      }
-
-      // Else setting all the even listners and resume content stream fetching.
-      file.content.on('data', (chunk: Buffer) => {
-        currentContentSize += chunk.length;
-
-        // If content size exceeds the max size limit, immediate stop further stream reading.
-        if (currentContentSize > maxSizeInBytes) {
-          console.info(`Content stream reached ${currentContentSize} bytes which is greater than the ${maxSizeInBytes} bytes limit.`);
-          fetchResult.code = FetchResultCode.MaxSizeExceeded;
+    let iterator: AsyncIterator<Buffer>;
+    try {
+      iterator = this.node.cat(hash);
+    } catch (e) {
+      // when an error is thrown, that means the hash points to something that is not a file
+      console.debug(`Error thrown while downloading content from IPFS for CID ${hash}: ${JSON.stringify(e, Object.getOwnPropertyNames(e))}`);
+      return { code: FetchResultCode.NotAFile };
```

Comment:
These are the errors it can throw so yes. If the CID is a directory, it is not a file, and if it has no content, it also is not a file

`    if (file.unixfs && file.unixfs.type.includes('dir')) {
      throw new Error('this dag node is a directory')
    }

    if (!file.content) {
      throw new Error('this dag node has no content')
    }`